### DISCUSSION TITLE
MQTT losing connection

### DISCUSSION THREAD

**Discussion by redfusion83** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#599] [Q&A]
I have implemented the controller with MQTT in Home Assistant. Sometimes it loses connection, all entities become unavailable and the buttons and commands stop working. When I then open the web interface, it usually loads fine, but all variables show NaN. After reloading the web interface everything works fine and also the MQTT connection comes back. However, this is not possible when I'm not at home. Do you have any idea how to solve this?

**Comment by visualapproach** [comment]
Try v4.2.1 beta if you haven't.
